### PR TITLE
Sort array multithread

### PR DESCRIPTION
**Pull request by zhurou603**
数组多线程排序，类似与矩阵乘法（将二维矩阵看成一维，分块进行乘法运算）
这里将一个数组切割为3块（4核笔记本）略小于物理核，尚未统计性能
分线程分别完成排序（利用算法库的sort函数）主线程完成分块后较大的数组排序
等待所有线程完成排序后，进行merge

改进：与bitonic sort进行对比以及性能分析

### CODE REVIEW

File: src/sort_array_multithread/sort_array_multithread.cpp [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+#include<limits>
+
+using namespace std;
+
+const int thread_num = 3;
+int data[10000];
+int length = sizeof(data) / sizeof(int);
+
+void create_data(int* data, int length){
+	srand((int)time(0));
```
Comment (wakensky) [inline src/sort_array_multithread/sort_array_multithread.cpp:17]: 上次电话聊到过 ： std::random_engine versus rand() in libc 的差异。  有找资料看一下吗？

File: src/sort_array_multithread/sort_array_multithread.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	int* index = new int[thread_num + 1];
+	for(int i = 0; i < thread_num + 1; i++){
+		index[i] = i * chunk_size;
+	}
+	
+	for(int i = 0; i < thread_num; i++){
+		sort_future[i].wait();	
+	}
+
+	delete[] sort_future;
```
Comment (wakensky) [inline src/sort_array_multithread/sort_array_multithread.cpp:58]: delete[] seems good

File: src/sort_array_multithread/sort_array_multithread.cpp [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	for(int i = 0; i < thread_num; i++){
+		sort_future[i] = sort_promise[i].get_future();
+	}
+
+	for(int i = 0; i < thread_num; i++){
+		thread sort_thread([=]{
+			sort(data + i*chunk_size, data + (i+1)*chunk_size);
+			sort_promise[i].set_value();
+		});
+		sort_thread.detach();
```
Comment (wakensky) [inline src/sort_array_multithread/sort_array_multithread.cpp:43]: detach 在这个局部的使用是合理的

File: src/sort_array_multithread/sort_array_multithread.cpp [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+	for(int i = 0; i < thread_num; i++){
+		sort_future[i].wait();	
+	}
+
+	delete[] sort_future;
+	delete[] sort_promise;
+
+	for(int i = 0; i < length; i++){
+		int min_index;
+		int min_num = numeric_limits<int>::max();
```
Comment (wakensky) [inline src/sort_array_multithread/sort_array_multithread.cpp:63]: 赞 numeric_limits的使用；  by the way, 告诉我 std::numeric_limits<float>::min() 和 lowest() 的区别
